src/utils/sources-tree/addToTree.js

```diff
@@ -150,10 +160,18 @@ export function addToTree(
   const url = getURL(source.get("url"), debuggeeUrl);
   const debuggeeHost = getDomain(debuggeeUrl);
 
-  if (isInvalidUrl(url, source)) {
+  if (isInvalidUrl(url, source) || !isUnderRoot(url, projectRoot)) {
     return;
   }
 
-  const finalNode = traverseTree(url, tree, debuggeeHost);
+  const finalNode = traverseTree(url, tree, debuggeeHost, projectRoot);
   finalNode.contents = addSourceToNode(finalNode, url, source);
 }
+
+function isUnderRoot(url, projectRoot) {
+  if (!projectRoot) {
+    return true;
+  }
+
+  return `/${url.group}${url.path}`.startsWith(projectRoot);
```

Comment:
Not quite sure I'd expect anything but a true/false from an isSomething function